lib/Twitter/System/Session.vala

```diff
@@ -56,6 +55,11 @@ public partial class Backend.Twitter.Session : AsyncInitable {
                                   server.client_secret,
                                   @"https://api.$(server.domain)/2/");
     proxy.access_token = access_token;
+    // Work around librest doing a DateTime comparison with the default non-expiring token
+    // 100 years should be more than enough for anyone!
+    var now = new GLib.DateTime.now();
+    var far_future = now.add_years (100);
+    proxy.set_expiration_date (far_future);
```

Comment:
With Twitter's access token, I previously had the issue that the token would actually "expire" after some point. So, in the old `main` branch, I had code to [get a new access token](https://github.com/CodedOre/NewCaw/blob/67042def3b2b592dfede844588cfabcaa4d6d285/lib/Twitter/System/Account.vala#L156-L161) and [saves that](https://github.com/CodedOre/NewCaw/blob/67042def3b2b592dfede844588cfabcaa4d6d285/src/System/Session.vala#L108-L109) on each startup.

I thought that this was just a thing of the new API, but it seems this bit of code seems to do the trick as well!

Reply:
I'm not sure. I keep getting a "Forbidden" every so often, and then Cawbird crashes and loses my Twitter account. I only noticed it after making this PR, so I've not looked in to it yet.

The date won't fix the Twitter expiry problem. It's purely because `librest` tries to compare the timestamps and the token doesn't have an expiration date set in the property. I thought I'd read something about the token being "perpetual" or "non-expiring", so I assumed it didn't have a date and `librest` was just failing to handle that. But I can't find where I read it. And other authentication issues suggest that that isn't the case.

Reply:
I'm still trying to work out this new flow. Does that mean that Twitter doesn't actually give an access token, but gives us a refresh token?

And were you storing it in `access_token` because that's the variable that we have? And it's the token that we need to track because it grants access to stuff (even if it's not the actual "access token")? I think that was part of what confused me about your `refresh_access_token` calls when I'd looked at other branches.

Reply:
I've tried putting the refresh lines into the `init_async` on the Session. But I get a segfault that appears to be from the refreshed date not having a timezone. Even though I don't do anything with the timestamps between the refresh and the point that it errors.

```
#0  0x00007f5330c47be4 in g_time_zone_get_offset () at /lib64/libglib-2.0.so.0
#1  0x00007f5330bfbf7a in g_date_time_difference () at /lib64/libglib-2.0.so.0
#2  0x00007f5330bfbff9 in g_date_time_compare () at /lib64/libglib-2.0.so.0
#3  0x00007f5330a87fc8 in  () at /lib64/librest-1.0.so.0
#4  0x00007f5330a83207 in  () at /lib64/librest-1.0.so.0
#5  0x00007f5330a84637 in rest_proxy_call_invoke_async () at /lib64/librest-1.0.so.0
#6  0x00007f5330f9262b in backend_server_call_co (_data_=0x558f63597d30) at ../lib/Base/System/Server.vala:76
#7  0x00007f5330f92489 in backend_server_call (self=0x558f5f85bea0, call=0x558f638f58a0, cancellable=0x0, _callback_=0x7f5330fb35d0 <backend_twitter_session_init_async_ready>, _user_data_=0x558f5f742c70)
    at ../lib/Base/System/Server.vala:38
#8  0x00007f5330fb3e97 in backend_twitter_session_real_init_async_co (_data_=0x558f5f742c70) at ../lib/Twitter/System/Session.vala:112
#9  0x00007f5330fb356a in backend_twitter_session_real_init_async (self=0x558f635a5200, io_priority=0, cancellable=0x0, _callback_=0x7f5330f93b78 <backend_session_from_data_ready>, _user_data_=0x558f6348db70)
    at ../lib/Twitter/System/Session.vala:29
```

Reply:
Slightly concerned that the logged time is `2123-01-14 @ 38947:37:03`. That's using `%T`. Which should be `%H:%M:%S`. So I don't have a clue how it gets to 38947 as a number of hours!

Reply:
IIRC, the old code in `main` worked that way:
After the authentication, Twitter gives us an refresh and an access token. The access token is used and saved as known. However, at the next run of the program, the saved access token is used as the refresh token to acquire a new access token.

To be honest, after a few months not working on NewCaw, even I don't really understand why and how this works...